### DISCUSSION TITLE
Problemas ao gerar as Migrations

### DISCUSSION THREAD

**Discussion by RanieryCG** [balta-io/7196#20] [Perguntas e Respostas] [answered]
Olá, tudo bem?

Eu estou acompanhando o curso, mas resolvi fazer com o .NET 6.

Ao tentar gerar as migrations eu estou tendo o seguinte erro: 

![image](https://user-images.githubusercontent.com/6059830/138498635-ac4f8630-96e6-419a-8f90-c0be335f4f52.png)

Essa é a estrutura que eu usei

![image](https://user-images.githubusercontent.com/6059830/138498706-baf53aa9-76e5-429f-93f7-b063bb30d6a4.png)

O projeto Organizacao.Api referencia os projetos: Organizacao.Dominio e Organizacao.Infraestrutura

O projeto Organizacao.Infraestrutura referencia os projetos: Organizacao.Dominio

No projeto Organizacao.API e Organizacao.Infraestrutura foram instalados os seguintes pacotes: 

- Microsoft.EnitityFrameworkCore.Design, 
- Microsoft.EnitityFrameworkCore.Tools, 
- Microsoft.EnitityFrameworkCore.Design 
- Microsoft.EnitityFrameworkCore.SqlServer.

No projeto Organizacao.API o serviço foi adicionado como na imagem abaixo:
![image](https://user-images.githubusercontent.com/6059830/138499311-c0424f65-2eff-4db8-85fe-f1adfe90e55e.png)

No projeto Organiacao.Infraestrutura o DbContext está assim:

![image](https://user-images.githubusercontent.com/6059830/138499394-c8d58f9d-213e-4a2e-bd1e-2cd1271bb1d8.png)

O camando rodado foi rodado dentro do Organizcao.Infraestrutura:
dotnet ef migrations add OrganizacaoInicial --startup-project ..\Organizacao.API\

**Comment by andrebaltieri** [comment, marked as answer]
@RanieryCG ele não consegue criar uma instância do seu data context por que ele não tem um construtor sem parâmetros. Como as migrations são resolvidas fora do contexto do ASP.NET, o `option` que você pede não pode ser resolvido.

Você pode criar um construtor sem parâmetros e informar uma conection string para ele funcionar.
